### PR TITLE
Add preact-vite framework and templates

### PR DESCRIPTION
**Pull request by tmeasday**
Telescoping on https://github.com/storybookjs/storybook/pull/19431

- [x] Add `preact-vite` framework
- [x] Add `preact-vite/x` templates. 

This is not tested yet.

### CODE REVIEW

File: code/frameworks/preact-vite/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+# Storybook for React
```
Comment (shilman) [inline code/frameworks/preact-vite/README.md:1]:
```suggestion
# Storybook for Preact
```

File: code/frameworks/preact-vite/src/typings.d.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// TODO: Replace, as soon as @types/react-dom 17.0.14 is used
+// Source: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fb0f14b7a35cde26ffaa82e7536c062e593e9ae6/types/react-dom/client.d.ts
+declare module 'react-dom/client' {
```
Comment (shilman) [inline code/frameworks/preact-vite/src/typings.d.ts:3]: Not sure whether this applies to preact? Above my pay grade 🤷

**Review by shilman — COMMENTED** [review]
Generally looks good, though I suspect there's too much react stuff in the framework ... For example I'd imagine that Preact has its own TS types.

File: code/frameworks/preact-vite/src/typings.d.ts (hunk 2 above)
Comment (tmeasday) [reply code/frameworks/preact-vite/src/typings.d.ts]: I simplified back to the `preact-webpack5` typings.

**Comment by tmeasday** [comment]
Closing but keeping the test here: https://github.com/storybookjs/storybook/pull/20542

**Pull request closed without merging**